components/net/http_loader.rs

```diff
@@ -316,18 +286,22 @@ impl Read for StreamedResponse {
 }
 
 impl StreamedResponse {
-    fn from_http_response(response: WrappedHttpResponse) -> io::Result<StreamedResponse> {
-        let decoder = match response.content_encoding() {
-            Some(Encoding::Gzip) => {
-                Decoder::Gzip(GzDecoder::new(response)?)
-            }
-            Some(Encoding::Deflate) => {
-                Decoder::Deflate(DeflateDecoder::new(response))
-            }
-            Some(Encoding::EncodingExt(ref ext)) if ext == "br" => {
-                Decoder::Brotli(Decompressor::new(response, 1024))
-            }
-            _ => {
+    fn from_http_response(response: HyperResponse) -> io::Result<StreamedResponse> {
+        let decoder = {
+            let headers = response.headers.clone();
```

Comment:
Cloning all of the headers associated with this response is inefficient. Let's do `headers.get::<ContentEncoding>().cloned()` instead.

Reply:
@jdm If I change to 
```rust
impl StreamedResponse {
    fn from_http_response(response: HyperResponse) -> io::Result<StreamedResponse> {
        let decoder = {
            let contentEncoding = response.headers.get::<ContentEncoding>().clone();
            if let Some(ref encoding) = contentEncoding {
                if encoding.contains(&Encoding::Gzip) {
                    Decoder::Gzip(GzDecoder::new(response)?)
```
Error again:
```bash
error[E0505]: cannot move out of `response` because it is borrowed
   --> components/net/http_loader.rs:294:50
    |
291 |             let contentEncoding = response.headers.get::<ContentEncoding>().clone();
    |                                   ---------------- borrow of `response.headers` occurs here
...
294 |                     Decoder::Gzip(GzDecoder::new(response)?)
    |                                                  ^^^^^^^^ move out of `response` occurs here
```

Reply:
```

if change to 
```rust
  let contentEncoding = response.headers.clone().get::<ContentEncoding>();
```
the lift time will be not long enough.

Reply:
I did say `cloned` instead of `clone`. Does that make a difference?

Reply:
sorry :(
I didn't read carefully.